### ISSUE TITLE
Error when editing Archetypes Richtext in Plone 5.2

### ISSUE THREAD

**Issue by pbauer** [plone/Products.CMFPlone#2832]
Editing a AT-based frontpage of Plone 5.2 results in:

```
2019-04-12 22:43:19,607 ERROR   [Zope.SiteErrorLog:250][waitress] 1555101799.60.836646791349 http://0.0.0.0:8080/Plone2/a-page/atct_edit
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 142, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 295, in publish_module
  Module ZPublisher.WSGIPublisher, line 229, in publish
  Module ZPublisher.mapply, line 85, in mapply
  Module ZPublisher.WSGIPublisher, line 57, in call_object
  Module Products.CMFFormController.FSControllerPageTemplate, line 92, in __call__
  Module Products.CMFFormController.BaseControllerPageTemplate, line 33, in _call
  Module Shared.DC.Scripts.Bindings, line 330, in __call__
  Module Shared.DC.Scripts.Bindings, line 367, in _bindAndExec
  Module Products.CMFCore.FSPageTemplate, line 248, in _exec
  Module Products.CMFCore.FSPageTemplate, line 188, in pt_render
  Module Products.PageTemplates.PageTemplate, line 83, in pt_render
  Module zope.pagetemplate.pagetemplate, line 135, in pt_render
  Module Products.PageTemplates.engine, line 90, in __call__
  Module z3c.pt.pagetemplate, line 173, in render
  Module chameleon.zpt.template, line 306, in render
  Module chameleon.template, line 209, in render
  Module chameleon.template, line 187, in render
  Module 7209f2a6812faa59836812eb2fdba964.py, line 101, in render
  Module 274ec22df352aefcc26d3db54b5d9607.py, line 913, in render_master
  Module a79477ce0c5096453d8a76064bc5786b.py, line 1028, in render_master
  Module a79477ce0c5096453d8a76064bc5786b.py, line 380, in render_content
  Module 274ec22df352aefcc26d3db54b5d9607.py, line 903, in __fill_main
  Module 274ec22df352aefcc26d3db54b5d9607.py, line 154, in render_main
  Module bb1927742a112c0ef6684ccf02c63a34.py, line 440, in render_body
  Module 6e85598a52bd6c2085aa9ed9d891cfe6.py, line 146, in render_edit
  Module e32b2b49e83985f7bffd77f84c02be2a.py, line 566, in render_edit
  Module 6e85598a52bd6c2085aa9ed9d891cfe6.py, line 113, in __fill_widget_body
  Module Products.Archetypes.Widget, line 1475, in edit
UnicodeEncodeError: 'ascii' codec can't encode character u'\xa0' in position 1142: ordinal not in range(128)

 - Expression: "error_id"
 - Filename:   ... es/Products/Archetypes/skins/archetypes/widgets/field.pt
 - Location:   (line 124: col 28)
 - Source:     tal:content="error_id" i18n:translate="">Validation Error</d ...
                            ^^^^^^^^
 - Expression: "field_macro | context/widgets/field/macros/edit"
 - Filename:   ... Products/Archetypes/browser/templates/patterns_widget.pt
 - Location:   (line 17: col 26)
 - Source:     ... se-macro="field_macro | context/widgets/field/macros/edit">
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Expression: "python:context.widget(field.getName(), mode='edit')"
 - Filename:   ... ypes/Products/Archetypes/skins/archetypes/edit_macros.pt
 - Location:   (line 106: col 49)
 - Source:     ... cro="python:context.widget(field.getName(), mode='edit')" />
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Expression: "body_macro"
 - Filename:   ... types/Products/Archetypes/skins/archetypes/base_edit.cpt
 - Location:   (line 85: col 35)
 - Source:     <metal:use_body use-macro="body_macro" />
                                          ^^^^^^^^^^
 - Expression: "context/main_template/macros/master"
 - Filename:   ... types/Products/Archetypes/skins/archetypes/base_edit.cpt
 - Location:   (line 36: col 23)
 - Source:     ... etal:use-macro="context/main_template/macros/master"
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Expression: "here/base_edit/macros/master"
 - Filename:   ... oducts/ATContentTypes/skins/ATContentTypes/atct_edit.cpt
 - Location:   (line 1: col 29)
 - Source:     ... n metal:use-macro="here/base_edit/macros/master" />
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ImplicitAcquisitionWrapper atct_edit at 0x1161c0e10>
               modules: <_SecureModuleImporter - at 0x10f497dd0>
               here: <ImplicitAcquisitionWrapper a-page at 0x116037eb0>
               wrapped_repeat: <SafeMapping - at 0x1179e3f80>
               user: <ImplicitAcquisitionWrapper - at 0x1160205a0>
               nothing: <NoneType - at 0x10cf956e8>
               container: <ImplicitAcquisitionWrapper Plone2 at 0x1172408c0>
               default: <object - at 0x10d086e90>
               root: <ImplicitAcquisitionWrapper  at 0x117362960>
               request: <WSGIRequest - at 0x1185181d0>
               traverse_subpath: <list - at 0x1179e3c68>
               loop: {...} (3)
               context: <ImplicitAcquisitionWrapper a-page at 0x116037eb0>
               translate: <function translate at 0x118c15f50>
               macroname: master
               options: {...} (2)
               target_language: <NoneType - at 0x10cf956e8>
```

The reason is in `Products.Archetypes.Widget, line 1475`. 

```python
# Render the combined widget
rendered = '{}\n{}'.format(
    textarea_widget.render(),
    etree.tostring(mt_select)
)
```

It should work with either `u'{}\n{}'` or `safe_encode(textarea_widget.render())`

**Comment by pbauer** [comment]
@mauritsvanrees maybe related to https://github.com/plone/Products.Archetypes/pull/128?

**Comment by mauritsvanrees** [comment]
Confirmed, but it is not related to my change. Locally I reverted Archetypes to the latest release, 1.15.5 from November, and had the same problem.

For clarity, instructions to reproduce:
* coredev 5.2, Python 2.7
* bin/instance-archetypes fg
* Create a Plone Site
* Install Archetypes Content Types for Plone
* Add a page with only ascii characters in the the rich text, save it, view it, edit it again: no problems.
* Edit that page and add a a non-ascii character in the rich text field and save it.
* View is fine.
* Edit it and you get the above traceback.

Both your proposals work when I try them. Turning it into unicode (text) seems best, as most of the code has that. I will make a PR.
